src/Language/PureScript/CodeGen/JS/Common.hs

```diff
@@ -75,107 +81,166 @@ identCharToString c = '$' : show (ord c)
-              , "private"
-              , "protected"
-              , "public"
-              , "return"
-              , "short"
-              , "static"
-              , "super"
-              , "switch"
-              , "synchronized"
-              , "this"
-              , "throw"
-              , "throws"
-              , "transient"
-              , "try"
-              , "typeof"
-              , "var"
-              , "void"
-              , "volatile"
-              , "while"
-              , "with"
-              , "yield" ] || properNameIsJsReserved name
-
-moduleNameToJs :: ModuleName -> String
-moduleNameToJs (ModuleName pns) =
-  let name = intercalate "_" (runProperName `map` pns)
-  in if properNameIsJsReserved name then "$$" ++ name else name
+  name `elem` jsAnyReserved
 
 -- |
--- Checks whether a proper name is reserved in Javascript.
+-- Checks whether a name matches a built-in value in Javascript.
 --
-properNameIsJsReserved :: String -> Bool
-properNameIsJsReserved name =
-  name `elem` [ "Infinity"
-              , "NaN"
-              , "Object"
-              , "Function"
-              , "Boolean"
-              , "Error"
-              , "EvalError"
-              , "InternalError"
-              , "RangeError"
-              , "ReferenceError"
-              , "SyntaxError"
-              , "TypeError"
-              , "URIError"
-              , "Number"
-              , "Math"
-              , "Date"
-              , "String"
-              , "RegExp"
-              , "Array"
-              , "Int8Array"
-              , "Uint8Array"
-              , "Uint8ClampedArray"
-              , "Int16Array"
-              , "Uint16Array"
-              , "Int32Array"
-              , "Uint32Array"
-              , "Float32Array"
-              , "Float64Array"
-              , "ArrayBuffer"
-              , "DataView"
-              , "JSON"
-              , "Intl" ]
+nameIsJsBuiltIn :: String -> Bool
+nameIsJsBuiltIn name =
+  elem name
+    [ "arguments"
+    , "Array"
+    , "ArrayBuffer"
+    , "Boolean"
+    , "DataView"
+    , "Date"
+    , "decodeURI"
+    , "decodeURIComponent"
+    , "encodeURI"
+    , "encodeURIComponent"
+    , "Error"
+    , "escape"
+    , "eval"
+    , "EvalError"
+    , "Float32Array"
+    , "Float64Array"
+    , "Function"
+    , "Generator"
+    , "GeneratorFunction"
+    , "Infinity"
+    , "Int16Array"
+    , "Int32Array"
+    , "Int8Array"
+    , "InternalError"
+    , "Intl"
+    , "isFinite"
+    , "isNaN"
+    , "JSON"
+    , "Map"
+    , "Math"
+    , "NaN"
+    , "Number"
+    , "Object"
+    , "parseFloat"
+    , "parseInt"
+    , "Promise"
+    , "Proxy"
+    , "RangeError"
+    , "ReferenceError"
+    , "Reflect"
+    , "RegExp"
+    , "Set"
+    , "SIMD"
+    , "String"
+    , "Symbol"
+    , "SyntaxError"
+    , "TypeError"
+    , "Uint16Array"
+    , "Uint32Array"
+    , "Uint8Array"
+    , "Uint8ClampedArray"
+    , "undefined"
+    , "unescape"
+    , "uneval"
```

Comment:
Yeah, probably not; we'll never cover every possible non-standard value in every browser. I'll take them out.